### PR TITLE
mixed chaos

### PR DESCRIPTION
**Pull request by jschueller**
attempt to solve #1333
- first implement bounds in Hyperbolic & norm inf enumerates, TODO: linear ? not easy...
- then in OrthogonalProductPolynomialFactory we set the upper bounds of discrete variables multi indices components to the size of the measure support minus 1

```
OrthogonalProductPolynomialFactory::buildMeasure()
{
  const UnsignedInteger size = coll_.getSize();
  Collection<Distribution> distributions(size);

  // get default upper bound
  Indices upperBound(LinearEnumerateFunction(size).getUpperBound());
  Bool hasDiscrete = false;

  for (UnsignedInteger i = 0; i < size; ++i)
  {
    distributions[i] = coll_[i].getMeasure();
    if (distributions[i].isDiscrete())
    {
      hasDiscrete = true;
      if (distributions[i].getSupport().getSize())
        upperBound[i] = distributions[i].getSupport().getSize() - 1;
      else
        throw InvalidArgumentException(HERE) << "Measure support is empty";
    }
  }
  measure_ = ComposedDistribution(distributions);

  // the enumerate function must support bounds
  if (hasDiscrete)
    phi_.setUpperBound(upperBound);
```

### CODE REVIEW

**Comment by jschueller** [comment]
@regislebrun turns out it was N-1:
```
TypeError: InvalidArgumentException : Error: cannot build a Krawtchouk polynomial of index greater than its first parameter n. Here, n=2 and you are trying to build the 3th polynomial.
```

**Comment by regislebrun** [comment]
@jschueller If the support has N points, the space dimension is N: the polynomials start at degree 0 and stop at degree N-1

File: python/src/EnumerateFunctionImplementation_doc.i.in [hunk 1]
```diff
@@ -288,3 +288,30 @@ Examples
...
+
+// ---------------------------------------------------------------------
+
+%define OT_EnumerateFunction_getUpperBound_doc
+"Accessor to the upper bound.
+
+Returns
+-------
+ub : sequence of int
+    Upper bound of the indices."
```
Comment (regislebrun) [inline python/src/EnumerateFunctionImplementation_doc.i.in:314]: Here we should say that the upper bound is included into the enumeration, e.g. with a bound equal to 4 a polynomial up to degree 4 will be generated.

**Review by regislebrun — APPROVED** [review]
Only a minor remark regarding the documentation of the bounds

File: python/src/EnumerateFunctionImplementation_doc.i.in (hunk 1 above)
Comment (jschueller) [reply python/src/EnumerateFunctionImplementation_doc.i.in]: ok, added '(includsive)'

**Review by sofianehaddad — APPROVED** [review]

**Pull request merged**